microsoft-edge/devtools-guide-chromium/evaluate-performance/index.md

```diff
@@ -183,22 +176,58 @@ This article gives you a lot to learn. But now you have a solid foundation in th
 
 Using the workflows and tools that you just learned, click **Optimize** on the demo to turn on the optimized code, take another performance recording, and then analyze the results.  From the improved framerate to the reduction in events in the flame chart in the **Main** section, the optimized version of the app does much less work, resulting in better performance.
 
-The optimized code uses a different sequence of actions to do less work. Note that this code could be made even faster by only using properties that only affect compositing, instead of manipulating the `top` property of every icon.
-<!--  > For more information, see [Use transform and opacity changes for animations](https://web.dev/stick-to-compositor-only-properties-and-manage-layer-count/#use-transform-and-opacity-changes-for-animations). todo: add rendering section when available -->
-
+Compare this snippet of JavaScript from the unoptimized version of the app:
+
+```javascript
+var pos = m.classList.contains("down")
+  ? m.offsetTop + distance
+  : m.offsetTop - distance;
+if (pos < 0) pos = 0;
+if (pos > maxHeight) pos = maxHeight;
+m.style.top = pos + "px";
+if (m.offsetTop === 0) {
+  m.classList.remove("up");
+  m.classList.add("down");
+}
+if (m.offsetTop === maxHeight) {
+  m.classList.remove("down");
+  m.classList.add("up");
+}
+```
+
+`m` references each blue icon on the page. In the unoptimized version, we refer to `pos` as the new vertical position of the icon, calculated based on `m.offsetTop + distance` if the icon is moving down and where `distance` is a fixed increment.  If the icon is moving up instead, the new position is instead calculated with `m.offsetTop - distance`.
+
+We check if we've exceeded either `0` or `maxHeight` as the new position whether we're moving up or down, adjust accordingly, and then we finally set the new position for each icon in CSS with `m.style.top = pos + "px";`.
+
+The optimized code uses a different sequence of actions to do less work. Here is the same snippet of JavaScript from the optimized version of the app: 
+
+```javascript
+var pos = parseInt(m.style.top.slice(0, m.style.top.indexOf("px")));
+m.classList.contains("down") ? (pos += distance) : (pos -= distance);
+if (pos < 0) pos = 0;
+if (pos > maxHeight) pos = maxHeight;
+m.style.top = pos + "px";
+if (pos === 0) {
+  m.classList.remove("up");
+  m.classList.add("down");
+}
+if (pos === maxHeight) {
+  m.classList.remove("down");
+  m.classList.add("up");
+}
+```
+
+In the optimized version, `pos` is used instead of `m.offsetTop`. The remaining work is the same but this snippet is faster than the previous one because we are no longer querying the style of each icon while also changing it, which triggers the browser to re-layout each icon.  
```

Comment:
```suggestion
In the optimized version, we first set the value of the `pos` variable by reading `m.style.top` instead of using `m.offsetTop`. Using the element's inline style is faster because reading `m.offsetTop` forces the browser engine to know where all the elements are on the page, which requires the engine to recalculate the styles and the layout.

We then set the new position of the icon like the previous version, but we don't read `m.offsetTop` again like we did before to adjust the icon's direction.

The unoptimized code reads and writes the position of the icon from two different places, forcing the browser to recalculate the style and layout on each frame. The optimized version, however, writes and reads the position of the icon in the inline styles only.
```

I'm not completely sure if this is clear to readers, let me know what you think.
In any case, your  understanding was correct. So feel free to use my suggestions or not.

Again, I think it'd be great to recreate this demo, and make the optimized version even more optimized so as to make more obviously different from the unoptimized version. Right now they're quite confusingly similar.

Reply:
At a glance, it looks like this addresses my previous ask (a few months ago), for a direct positive description of what the optzn is.